src/coins.cpp

```diff
@@ -199,7 +206,14 @@ bool CCoinsViewCache::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlockIn
             } else {
                 // A normal modification.
                 cachedCoinsUsage -= itUs->second.coin.DynamicMemoryUsage();
-                itUs->second.coin = std::move(it->second.coin);
+                if (erase) {
+                    // The `move` call here is purely an optimization; we rely on the
+                    // `mapCoins.erase` call in the `for` predicate to actually remove
```

Comment:
nit: the third part of the for loop is an expression, not a predicate (more precisely, the _increase_ or _iteration_ expression). A predicate is a function which returns true/false.